### DISCUSSION TITLE
WARNING: Dataset not found, nonexistent paths: ['/content/valid/images']

### DISCUSSION THREAD

**Discussion by ghost** [ultralytics/yolov5#6381] [Q&A]
I have read the previous post regarding this issue but the solution, pip install -U roboflow is not working for me.

**My dataset location:** 

%cat /content/yolov5/data.yaml
train: ../train/images
val: ../valid/images

**The taining command:**

%%time
%cd /content/yolov5/
!python train.py --img 416 --batch 16 --epochs 100 --data /content/yolov5/data.yaml --cfg /content/yolov5/models/custom_yolov5s.yaml --weights '' --name yolov5s --cache

**Errors:** 

/content/yolov5
github: ⚠️ WARNING: code is out of date by 787 commits. Use 'git pull' to update or 'git clone https://github.com/ultralytics/yolov5' to download latest.
YOLOv5 v4.0-126-g886f1c0 torch 1.10.0+cu111 CUDA:0 (Tesla T4, 15109.75MB)

Namespace(adam=False, batch_size=16, bucket='', cache_images=True, cfg='/content/yolov5/models/custom_yolov5s.yaml', data='/content/yolov5/data.yaml', device='', entity=None, epochs=100, evolve=False, exist_ok=False, global_rank=-1, hyp='data/hyp.scratch.yaml', image_weights=False, img_size=[416, 416], linear_lr=False, local_rank=-1, log_artifacts=False, log_imgs=16, multi_scale=False, name='yolov5s_fyp', noautoanchor=False, nosave=False, notest=False, project='runs/train', quad=False, rect=False, resume=False, save_dir='runs/train/yolov5s_fyp3', single_cls=False, sync_bn=False, total_batch_size=16, weights='', workers=8, world_size=1)
wandb: Install Weights & Biases for YOLOv5 logging with 'pip install wandb' (recommended)
Start Tensorboard with "tensorboard --logdir runs/train", view at http://localhost:6006/
hyperparameters: lr0=0.01, lrf=0.2, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0

WARNING: Dataset not found, nonexistent paths: ['/content/valid/images']
Traceback (most recent call last):
  File "train.py", line 531, in <module>
    train(hyp, opt, device, tb_writer, wandb)
  File "train.py", line 65, in train
    check_dataset(data_dict)  # check
  File "/content/yolov5/utils/general.py", line 145, in check_dataset
    raise Exception('Dataset not found.')
Exception: Dataset not found.
CPU times: user 40.7 ms, sys: 18.1 ms, total: 58.9 ms
Wall time: 5.14 s

**Comment by glenn-jocher** [comment]
@RealONE01 before you do anything you should update your code as it is out of date by almost 800 commits. Then head over to the **Train Custom Data** tutorial to get started:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://github.com/ultralytics/yolov5/issues/2518)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TFLite, ONNX, CoreML, TensorRT Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

Good luck 🍀 and let us know if you have any other questions!

**Comment by muhammad-faizan-122** [comment]
The issue is due to not found actual dataset path. I found same issue when i trained the Yolov5 model on custom dataset using google colab, I did the following to resolve this. 
1. Make sure provide correct path of data.yaml of dataset.
2. Make sure path of dataset in data.yaml should be be corrected.
3. train, test, and valid key should contain path with respect to the main path of the dataset.
Example data.yaml file given below.

path: /content/drive/MyDrive/car-detection-dataset
train: train/images
val: valid/images
test: test/images

nc: 1
names: ['car']

**Reply by yashika-git** [reply]
@muhammad-faizan-122, Can you please tell where should the labels for train and test images be stored?

**Reply by muhammad-faizan-122** [reply]
Dataset directory format should be following.
**Dataset**
   - train 
     - images
     - labels
   - test
     - images
     - labels 
   - valid
     - images
     - labels

labels directory contains the txt files, each txt file contains ground truth class of object with its bounding boxes parameters. The name of the image in images directory and its label in labels directory must be same.

**Reply by yashika-git** [reply]
Thanks @muhammad-faizan-122, the data.yaml file should also be in the Dataset folder?

**Reply by muhammad-faizan-122** [reply]
Yes.

**Reply by jiminpopo** [reply]
I change the path of dataset in data.yaml file. It works well

**Comment by rosevpattani** [comment]
The same problem with me too in yolov7.

WARNING: Dataset not found, nonexistent paths: ['/content/gdrive/MyDrive/Object/yolov7/content/gdrive/MyDrive/Object/yolov7/data/val/img']
Traceback (most recent call last):
  File "train.py", line 616, in <module>
    train(hyp, opt, device, tb_writer)
  File "train.py", line 97, in train
    check_dataset(data_dict)  # check
  File "/content/gdrive/MyDrive/Object/yolov7/utils/general.py", line 173, in check_dataset
    raise Exception('Dataset not found.')
Exception: Dataset not found.


Please give me suggestions

**Reply by yashika-git** [reply]
I resolved the issue by using absolute paths instead of relative paths and also by switching the type of slash (forward or backward) I was using in the path.

**Reply by kuan79223** [reply]
我也有相同的問題

**Reply by 15ihsanmuttaqin** [reply]
I'm also facing the same problem as you, has someone found a solution?

**Reply by kuan79223** [reply]
solution: re-download the author's package

**Reply by rajat1425** [reply]
i also facing the same issue on kaggle. Can anyone guide me how to resolve it?

**Comment by cbru241** [comment]
I encountered the same issue, but I resolved it by changing the dataset download from YOLOv5 Oriented Bounding Boxes to YOLO v5 PyTorch one PyTorch. Maybe this will help someone in the future :)

**Comment by jpinzonc** [comment]
![Screenshot 2023-01-28 at 4 16 36 PM](https://user-images.githubusercontent.com/10221668/215293575-3ab26a0c-8f1a-4142-af32-3d46d5a81180.png)

I am having the same issue, but the paths are correct. I have tried all the combinations above with no luck. I am using Google Colab.

**Reply by jpinzonc** [reply]
I fix it by editing the path inside the .yaml file, from: 

test: ../test/images
train: SoccerBallDetect-7/train/images
val: SoccerBallDetect-7/valid/images

to: 

test: ../test/images
train: /content/SoccerBallDetect-7/train/images
val: /content/SoccerBallDetect-7/valid/images

**Comment by rutchaphon123** [comment]
![fix](https://github.com/ultralytics/yolov5/assets/41964089/8983c4a0-c3b1-4240-a47f-e3cdc8b54a11)
Change directory in data.yaml by double click and Ctrl + S to save. It work for me.
